lib/src/widgets/chat/ds_end_calls_message_bubble.widget.dart

```diff
@@ -0,0 +1,272 @@
+                                        : DSIcons.voip_calling_outline
+                                    : DSIcons.voip_ended_outline,
+                                size: 20.0,
+                              ),
+                            ),
+                          ],
+                        ),
+                      ),
+                      Column(
+                        crossAxisAlignment: CrossAxisAlignment.start,
+                        children: [
+                          DSCaptionText(
+                            'calls.voice-text'.translate(),
+                            fontWeight: FontWeight.bold,
+                            color: _foregroundColor,
+                          ),
+                          DSCaptionSmallText(
+                            _isCallAnswered
+                                ? 'calls.answered'.translate()
+                                : 'calls.unanswered'.translate(),
+                            color: _foregroundColor,
+                          ),
+                        ],
+                      ),
+                    ],
+                  ),
+                ],
+              ),
+              Flexible(
+                child: Column(
+                  children: [
+                    FutureBuilder(
+                      future:
+                          content['identification'].toString().asPhoneNumber(),
+                      builder: (context, snapshot) {
+                        if (snapshot.connectionState == ConnectionState.done) {
+                          return DSCaptionSmallText(
+                            snapshot.data,
+                            color: _foregroundColor,
+                          );
+                        }
+                        return const DSSpinnerLoading();
+                      },
+                    )
+                  ],
+                ),
+              ),
+            ],
+          ),
+          if (_isCallAnswered && onAsyncFetchSession != null)
+            StreamBuilder(
+              stream: _streamController.stream,
+              builder: (_, __) {
+                return FutureBuilder<String?>(
+                  future: onAsyncFetchSession!(
+                    content['sessionId'],
+                  ),
+                  builder: (_, snapshot) {
+                    return switch (snapshot.connectionState) {
+                      ConnectionState.waiting => _buildLoading(),
+                      ConnectionState.done => snapshot.hasError
+                          ? _buildError()
+                          : snapshot.data?.isEmpty ?? true
+                              ? const SizedBox.shrink()
+                              : _buildAudioPlayer(snapshot.data!),
+                      _ => const SizedBox.shrink(),
+                    };
+                  },
+                );
+              },
+            )
+        ],
+      ),
+    );
+  }
+
+  Widget _buildContainer(final Widget child) => SizedBox(
```

Comment:
Since it is used in 3 places and receives a child widget as parameter, I highly recommend to turn this into an `Stateless Widget` instead of a method.